### PR TITLE
fix docs / proxy app

### LINKED ISSUE #2986: sync docs & tmint for nil/counter apps
Hello! 

I tried to follow the instruction from [this page](https://tendermint.com/docs/tendermint-core/using-tendermint.html#run) by using other in-process apps such as 'nil' or 'counter' but it shows errors as follows

image

Please help advice. Thank you.

### PR DESCRIPTION
**Pull request by zramsay**
closes #2986 

* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGELOG_PENDING.md

### CODE REVIEW

File: proxy/client.go [hunk 1]
```diff
@@ -64,13 +65,17 @@ func (r *remoteClientCreator) NewABCIClient() (abcicli.Client, error) {
...
-		return NewLocalClientCreator(kvstore.NewKVStoreApplication())
-	case "persistent_kvstore":
 		fallthrough
+	case "kvstore":
+		return NewLocalClientCreator(kvstore.NewKVStoreApplication())
 	case "persistent_dummy":
+		fallthrough
+	case "persistent_kvstore":
 		return NewLocalClientCreator(kvstore.NewPersistentKVStoreApplication(dbDir))
 	case "nilapp":
```
Comment (melekes) [inline proxy/client.go]: I don't like the name to be honest. Maybe rename to `empty` (or `noop`)

File: proxy/client.go [hunk 2]
```diff
@@ -64,13 +65,17 @@ func (r *remoteClientCreator) NewABCIClient() (abcicli.Client, error) {
 
 func DefaultClientCreator(addr, transport, dbDir string) ClientCreator {
 	switch addr {
-	case "kvstore":
-		fallthrough
+	case "counter":
+		return NewLocalClientCreator(counter.NewCounterApplication(false))
+	case "counter_serial":
+		return NewLocalClientCreator(counter.NewCounterApplication(true))
 	case "dummy":
```
Comment (melekes) [inline proxy/client.go]: I think it's time to remove `dummy/persistent_dummy`

File: proxy/client.go (hunk 1 above)
Comment (ebuchman) [reply proxy/client.go]: Great idea - let's call it `noop`. And we should deprecate `dummy` and `nillapp` names for v0.28

**Comment by zramsay** [comment]
comments addressed

File: abci/cmd/abci-cli/abci-cli.go [hunk 3]
```diff
@@ -58,7 +58,7 @@ var RootCmd = &cobra.Command{
 	PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
 
 		switch cmd.Use {
-		case "counter", "kvstore", "dummy": // for the examples apps, don't pre-run
+		case "counter", "kvstore": // for the examples apps, don't pre-run
```
Comment (melekes) [inline abci/cmd/abci-cli/abci-cli.go]:
```suggestion
		case "counter", "kvstore", "noop": // for the examples apps, don't pre-run
```

**Pull request merged**